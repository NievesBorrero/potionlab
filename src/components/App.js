import React from 'react';

import {APPLICATION_ROUTES} from '../constants/routes';
import RouterController from './RouterController';

export const App = () => {
  return (
        <div>
            <RouterController routes={APPLICATION_ROUTES} />
        </div>
  );
};
