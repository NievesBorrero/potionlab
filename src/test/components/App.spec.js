import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { routerMiddleware } from 'connected-react-router'

import App from '../../components/App'

const initialState = {}

const middlewares = [routerMiddleware]
const mockStore = configureStore(middlewares)

describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const store = mockStore(initialState)
    const component = shallow(
      <Provider store={store}>
        <App debug />
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })
})
