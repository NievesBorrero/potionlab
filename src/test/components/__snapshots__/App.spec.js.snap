// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <Component
    debug={true}
  />
</ContextProvider>
`;
